src/libServer/WebsocketServer.cpp

```diff
@@ -260,10 +282,47 @@ void WebsocketServer::on_message(const connection_hdl& hdl,
         }
         break;
       }
+      case TXNLOG: {
+        set<Address> track_addresses;
+        if (j_query.isMember("addresses") && j_query["addresses"].isArray() &&
+            !j_query["addresses"].empty()) {
+          for (const auto& addr_str : j_query["addresses"]) {
+            try {
+              if (!JSONConversion::checkStringAddress(addr_str.asString())) {
```

Comment:
In normal case where the address is valid, `asString` is called twice. Better to store the result in a local variable to be more efficient.